tt_metal/programming_examples/contributed/vecadd/vecadd.cpp

```diff
@@ -175,21 +184,25 @@ int main(int argc, char** argv) {
     SetRuntimeArgs(program, writer, core, {c->address(), n_tiles});
     SetRuntimeArgs(program, compute, core, {n_tiles});
 
-    // We have setup the program. Now we can queue the kernel for execution.
-    // The last argument to EnqueueProgram is a boolean that specifies whether
+    // We have setup the program. Now we can add it to the workload and queue it for execution.
+    // The last argument to EnqueueMeshWorkload is a boolean that specifies whether
     // we wait for the program to finish execution before returning. I've set
     // it to true. But alternatively, you can set it to false and call
     // `Finish(cq)` to wait for all programs to finish.
     // But it shouldn't matter in this case since we block on reading the output
     // buffer.
-    EnqueueProgram(cq, program, true);
+    distributed::AddProgramToMeshWorkload(workload, std::move(program), device_range);
+    distributed::EnqueueMeshWorkload(cq, workload, true);
     // Finish(cq);
     std::cout << "Kernel execution finished" << std::endl;
 
     // Read the output buffer.
     std::vector<uint32_t> c_data;
-    EnqueueReadBuffer(cq, c, c_data, true);
 
+    // We're reading from a shard allocated on Device Coordinate 0, 0, since this is a 1x1
+    //  When the MeshDevice is 2 dimensional, this API can be used to target specific physical devices
+    distributed::ReadShard(cq, c_data, c, device_coord);
```

Comment:
Question: Can we cann `EnqueueReadMeshBuffer`. The name makes much more sense - users will know that we are reading data back from device(s).